src/Jackett.Common/Indexers/BaseIndexer.cs

```diff
@@ -265,7 +265,7 @@ protected virtual IEnumerable<ReleaseInfo> FilterResults(TorznabQuery query, IEn
                 return results;
 
             var filteredResults = results.Where(
-                result => result.Category.IsEmptyOrNull() || query.Categories.Intersect(result.Category).Any() ||
+                result => result.Category?.Any() != true || query.Categories.Intersect(result.Category).Any() ||
```

Comment:
I know this one gave us trouble the last time I touched the `IsEmptyOrNull()` method, so here's the new version:

if `result.Category` is null, then the left side is null and `null != true` is true, `IsNull == true`
if `result.Category` is empty, then `result.Category.Any()` is false and `false != true` is true, `IsEmpty == true`

So this code is equivalent to `.IsEmptyOrNull()` and avoids any errors.

Reply:
This paragraph isn't about explaining my code, per se. The automated refactoring tools introduced a breaking change surrounding this line in #7312 which later had to be tracked down and fixed in #7338. Similarly, the automated refactoring tools say "merge sequential checks" for your proposed changes and convert it into the code in the commit. So the explanation is there to show that this time the automated tools are getting it right, and that I noticed it and considered it before continuing with the commit.

This also prevents anyone else from coming in behind us with an automated tool that will want to merge these two checks introducing a bug if it doesn't implement it correctly.